src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -756,4 +948,156 @@ private double divideOrReturnZero(double numerator, int denominator) {
+                });
+                perRecipientResponse.computeIfPresent(response.recipient, (key, responseCountPerOption) -> {
+                    // update responseCount here
+                    FeedbackMsqResponseDetails frd = (FeedbackMsqResponseDetails) response.getResponseDetails();
+                    return getNumberOfResponsesForEachResponse(frd, responseCountPerOption);
+                });
+            });
+            return perRecipientResponse;
+        }
+
+        /**
+         * Returns a HTML string which contains a sequence of "td" tags.
+         * The "td" tags have data related to a sub question.
+         * The sequence of "td" tags are not enclosed in a "tr" tag.
+         */
+        public String getPerRecipientStatsBodyFragmentHtml(String recipientEmail,
+                Map<String, Integer> recipientResponses, FeedbackSessionResultsBundle bundle) {
+            StringBuilder html = new StringBuilder(100);
+
+            List<String> cols = msqStats.generateStatisticsForEachRecipient(recipientEmail, recipientResponses, bundle);
+
+            // Generate HTML for all <td> entries using template
+            // Reuse Mcq result template until there is any reason to use a separate template.
+            for (String col : cols) {
+                html.append(
+                        Templates.populateTemplate(FormTemplates.MCQ_RESULT_RECIPIENT_STATS_BODY_ROW_FRAGMENT,
+                        Slots.MCQ_RECIPIENT_STAT_CELL, col));
+            }
+
+            return html.toString();
+        }
+
+        /**
+         * Returns a HTML string which contains a sequence of "tr" tags.
+         * The "tr" tags enclose a sequence of "td" tags which have data related to a sub question.
+         * The sequence of "tr" tags are not enclosed in a "tbody" tag.
+         */
+        public String getPerRecipientStatsBodyHtml(List<FeedbackResponseAttributes> responses,
+                FeedbackSessionResultsBundle bundle) {
+            StringBuilder bodyBuilder = new StringBuilder(100);
+            Map<String, Map<String, Integer>> perRecipientResponses = calculatePerRecipientResponseCount(responses);
+
+            for (Map.Entry<String, Map<String, Integer>> entry : perRecipientResponses.entrySet()) {
+                String recipient = entry.getKey();
+                Map<String, Integer> responsesForRecipient = entry.getValue();
+                String statsRow = getPerRecipientStatsBodyFragmentHtml(recipient, responsesForRecipient, bundle);
+                // Reuse Mcq result template until there is any reason to use a separate template.
+                bodyBuilder.append(Templates.populateTemplate(FormTemplates.MCQ_RESULT_RECIPIENT_STATS_BODY_FRAGMENT,
+                        Slots.MCQ_RECIPIENT_STAT_ROW, statsRow));
+            }
+
+            return bodyBuilder.toString();
+        }
+
+        /**
+         * Returns the 'Per Recipient' stats body part for CSV files.<br>
```

Comment:
Is `<br>` a typo here? :o

Reply:
Oops! Actually this is not a typo, there was a line after this where I guess I wrote `Most of the methods reuse McqStatistics to generate msq stats`, but I removed it, as it was kind of misleading, as only a few methods are reusing McqStatistics class. :sweat_smile: Will remove it in the next push.